src/citrine/resources/gemd_resource.py

```diff
@@ -106,18 +114,21 @@ def register_all(self, models: List[DataConcepts], *, dry_run=False) -> List[Dat
         for obj in models:
             by_type[obj.typ].append(obj)
         typ_groups = sorted(list(by_type.values()), key=lambda x: writable_sort_order(x[0]))
-        batch_size = 50
-        for typ_group in typ_groups:
-            num_batches = len(typ_group) // batch_size
-            for batch_num in range(num_batches + 1):
-                batch = typ_group[batch_num * batch_size: (batch_num + 1) * batch_size]
-                if batch:  # final batch is empty when batch_size divides len(typ_group)
-                    registered = self._collection_for(batch[0]) \
-                        .register_all(batch, dry_run=dry_run)
-                    for prewrite, postwrite in zip(batch, registered):
-                        if isinstance(postwrite, BaseEntity):
-                            prewrite.uids = postwrite.uids
-                    resources.extend(registered)
+
+        if status:
+            try:
+                from tqdm.auto import tqdm
+            except ImportError:  # pragma: no cover
+                raise ValueError('Display of a status bar requires '
```

Comment:
Why would we not just add this to the dependency list?

Reply:
Because it's not necessary for the main flow of the library and thus introduces an unnecessary point of failure in the dependency chain.

It could reasonably be argued this is a low-risk dependency.  I'm neutral, biased toward what I wrote, but would be easily persuaded.